modules/p2/ding_serendipity/ding_serendipity.module

```diff
@@ -488,6 +501,54 @@ function ding_serendipity_user_audience($account = NULL) {
   return $result;
 }
 
+/**
+ * Fetch random items, but with probability 50% among first 9% of items.
+ *
+ * @param array $list
+ *   The array to pick from.
+ * @param int $cnt
+ *   Number of items to pick.
+ *
+ * @return array
+ *   Array of randomly picked items.
+ */
+function ding_serendipity_pick_weighted_random(array $list, $cnt) {
+  $num = count($list);
+
+  if ($cnt >= $num) {
+    return $list;
+  }
+
+  // Pick cnt weighted random items. Collisions may occur.
```

Comment:
A description of the formula is in the docblock, I've copied that description down to this line for clarity.

Reply:
Sorry - I did not make myself clear. What I meant was:

For me there is no straight line from 50% probability for first 9% of items to `round((pow(100, $rnd) - 1) * ($num - 1) / 99)`.

Do you have a reference for the formula e.g. an external source which backs that the formula actually does what we want it? It would also be nice with a reference to what needs to change if we want to adjust the factors e.g. 50% probability for first 15% of items or 75% probability for first 9% of items.

Reply:
Description of the calculation is added to the upcoming PR. We can change the algortithm all together if some more public or common approach is prefered.